### ISSUE TITLE
Input values not properly applied after object update?

### ISSUE THREAD

**Issue by donnysim** [vuejs/vue#5502]
### Version
2.2.6

### Steps to reproduce
http://i.imgur.com/X2WsIYK.gifv

I can't reproduce it individually which implies there's something very specific but I can send you laravel repository for testing (privately).

Before modal opening the form fields object is set to empty object, which in appears to break it, but I still can't reproduce it in simple steps.

### What is expected?
Input values are kept and displayed.

### What is actually happening?
Input value disappear, but the object still contains them. The values are displayed only after $forceUpdate() but then the displayed values are saved until another $forceUpdate().

**Comment by posva** [comment]
You probably have something emptying the form.
Next time, please ask questions on the [forum](http://forum.vuejs.org/) or [StackOverflow](http://stackoverflow.com/tags/vue.js) first. 
BTW A gif is not a valid repro. You can share a repro in a repository but you must keep the strict minimal, so no Laravel.

**Comment by donnysim** [comment]
I don't have anything emptying the form, if instead of an empty object I pass object with fields already predefined with empty values, it works no problem, if it's an empty object, it fails to update when form watcher fires.

**Comment by donnysim** [comment]
In short, if I do:
```
this.clientForm.setData({
                    title: '',
                    company_code: '',
                    vat: '',
                    address: '',
                    city: '',
                    phone: '',
                    email: '',
                });
```
it works,
if I do:
```
this.clientForm.setData({});
```
it doesnt. When you inspect the object it, it has all the values - shown in the gif.

**Comment by donnysim** [comment]
If I try to debug it, the value disappears somewhere in the middle of `updateChildren`.

**Comment by donnysim** [comment]
Seems like I'll be able to replicate so I'll reopen a new issue once I have a cleaner repro.
